### PR TITLE
Refact added formik yup in form

### LINKED ISSUE #143: ✨ Add axios post method for creating lead in CRM
Cc: @pierssinclairssw @adamcogan @lukecookssw 

To Myself,
  
### Pain

Currently, on the booking form when you try to book, it doesn't do anything. We need to update it so it uses axios to call the old website endpoint for creating a lead in CRM.

### Tasks

- [x] Add axios post method to endpoint of the old website
- [x] Test creating lead on [CRM](https://ssw.crm6.dynamics.com/main.aspx?appid=05daa2a9-8768-446e-b9d3-580c8a6f9b8b&pagetype=entitylist&etn=lead&viewid=5a926b99-3a5f-df11-ae90-00155d2e3002&viewType=1039)
- [x] Redirect to Thankyou page after successful form submission

### Screenshots

Image
**Figure: The Submit button on the booking form doesn't do anything**

Thanks!

### LINKED ISSUE #194: ✨ Consulting - Add "I'm a developer" referral form
Cc: @lukecookssw @pierssinclairssw @ozairashfaqueSSW 

To Myself

### Describe the Pain

The form is showing the Booking form instead of the Referral form

image
Figure: What is currently being shown

### To Reproduce
1. Go to /consulting/are-you-stuck
2. Click on "I'm a Developer"
2. Click on "Refer SSW Consulting Services"
4. See error, the wrong booking form is shown

### Expected Behavior

This referral form should be shown instead
image
Figure: What should be showing when the referral button is clicked

### Tasks
- [x] Investigate the end point of each form
- [x] Build the form accordingly
- [x] Test

### Environment
 - Device: Any 
 - Browser: Any

### Screenshots

Thanks!

### LINKED ISSUE #198: 🟢 Subtask - reCAPTCHA
Cc: @adamcogan @pierssinclairssw @lukecookssw @bigtony97 @paulsmythssw @kevinpho2022 @christoment 

To Myself,
  
### Pain

Currently, [legacy website](https://www.ssw.com.au/ssw/) is using reCAPTCHA to prevent spam and abuse from entering the site while submitting forms.
### Suggested Solution

- reCAPTCHA can be injected through GTM, but it is not a suggested solution as GTM requires [Server-side tagging](https://developers.google.com/tag-platform/tag-manager/server-side/intro#:~:text=Server%2Dside%20tagging%20is%20a,user%20activity%20wherever%20it%20happens.). As @christoment advised not to use GTM for recaptcha.
- Inject reCAPTCHA script into the forms 
### Tasks

- [x] Investigate how to implement reCAPTCHA
- [x] Implement reCAPTCHA into forms
- [x] Test the form

### Screenshots

image
**Figure: reCAPTCHA prevents bots**
Thanks!

### LINKED ISSUE #201: ♻️ Adding Formik library for Booking form
Cc: @adamcogan @pierssinclairssw @lukecookssw @bigtony97 @paulsmythssw @kevinpho2022 

To Myself,

### Pain

Currently, in our bookingForm and shareForm, we are using HTML5 for form validation and error handling, which is not a good option. Therefore, we will be adding [Formik](https://formik.org/docs/overview) library into our forms.
### What code could be improved?

We need to refactor [bookingForm](https://frontdoorstaging.z01.azurefd.net/consulting/consulting-options/) component 
### Tasks

- [x] Add Formik to current form component
- [x] Handle errors and validation via Formik

### Screenshots

image
**Figure: BookingForm in consulting page**

Thanks!

### PR DESCRIPTION
**Pull request by ozair-ashfaque**
resolves #201 resolves #194 & resolves #198 

- Added Formik lib for form
- Added FormGroup input, textArea and select for rendering
- Added validation Schema using Yup lib
- Added recaptcha to the form 
- Added recaptcha env to env.template
- Added dynamically generating shareForm and bookingForm using same component
- Added form modal animation from top

bookingForm
**Figure: Are-You-Stuck page have booking and share form**

### CODE REVIEW

File: src/components/are-you-stuck/components/busi-booking/index.tsx [hunk 1]
```diff
@@ -10,24 +10,38 @@ const BusiBooking = () => {
...
       <section>
-        <h5 className={styles.firstLine}>We are enterprise software consultants who solve complex business problems and reduce risk, using best practices and the latest technologies.</h5>
+        <h5 className={styles.firstLine}>
+          We are enterprise software consultants who solve complex business
+          problems and reduce risk, using best practices and the latest
+          technologies.
+        </h5>
         <h5 className={styles.firstLine}>See how SSW can help:</h5>
-        <ul >
+        <ul>
```
Comment (JackDevAU) [inline src/components/are-you-stuck/components/busi-booking/index.tsx:19]:
```suggestion
        <ul className={styles.theList}>
```
Probably better to just add the styling to the parent component targeting the `<li>` elements below. then just add a separate style to the last item if needed.

File: src/components/bookingFormFormik/formData/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+export const FormSubmissionData = (values, isShareForm, contactReCaptcha) => {
+  console.log("values", values);
```
Comment (JackDevAU) [inline src/components/bookingFormFormik/formData/index.tsx:2]: we can comment this out or remove it

File: src/components/bookingFormFormik/formData/index.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "<br/>";
+
+  body =
+    body +
+    (isShareForm
+      ? `Referred By: ${values.fullName} (${values.email})`
+      : `Note: ${values.note}`) +
+    "<br/><br/>";
+
+  console.log(body);
```
Comment (JackDevAU) [inline src/components/bookingFormFormik/formData/index.tsx:40]: We can comment this out or remove it

File: src/components/bookingForm/index.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { Button, Col, Form } from "react-bootstrap";
 import axios from "axios";
 import "./index.css";
```
Comment (JackDevAU) [inline src/components/bookingForm/index.tsx:5]: Is this file still needed?

File: src/components/bookingFormFormik/index.tsx [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+    }
+  }, []);
+  useEffect(() => {
+    // every time isShowState changes, recreate the schema and set it in the state
+    setSchema(ValidationSchema(isShowStates, isShareForm));
+  }, [isShowStates]);
+
+  //Form Data
+
+  const handleActiveInputLabel = (targetInput, value) => {
```
Comment (JackDevAU) [inline src/components/bookingFormFormik/index.tsx:57]: Is this used for moving a forms input label? 

We should probably just look into using a UI lib for stuff like this 😲

File: src/components/bookingFormFormik/formData/index.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
+export const FormSubmissionData = (values, isShareForm, contactReCaptcha) => {
```
Comment (JackDevAU) [inline src/components/bookingFormFormik/formData/index.tsx:1]: Is this file needed? what are all the `<br />` tags for?

If it is needed. I would change all these string to string templates i.e. using the ` characters

File: src/components/bookingFormFormik/formGroupInput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import { VALIDATION_SUCCESS_MESSAGE } from "../../../constants";
+
+const FormGroupInput = ({
+  name,
+  inputType,
+  activeInput,
+  activeLabelClass,
+  handleChange,
+  isValid,
+  isErrors,
```
Comment (JackDevAU) [inline src/components/bookingFormFormik/formGroupInput/index.tsx:12]: import { useField } from "formik"
const [field, meta] = useField(props)

A small improvement - There's a formik hook that can bring in the field values and metadata (errors etc)

File: src/components/bookingFormFormik/index.tsx [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+        <div className="modal-body">
+          <div className="get-started-form">
+            <h2>{isShareForm ? SHARE_FORM_TITLE : CONTACT_FORM_TITLE}</h2>
+            <Formik
+              validationSchema={schema}
+              initialValues={InitialValues}
+              onSubmit={handleOnSubmit}
+            >
+              {({ handleSubmit, handleChange, values, touched, errors }) => (
+                <Form
```
Comment (JackDevAU) [inline src/components/bookingFormFormik/index.tsx:115]: I believe this is better, but we still pass so many props around 😲 some of these can be retrieved via the `useField` hook that comes with formik.

**Review by JackDevAU — APPROVED** [review]
Good work!

Some nice improvements from the last iteration, definitely moving in the right direction by using formik. Apart from my comments, these forms will still need some iterating on. There is still a lot of the validation and prop passing that can be avoided or simplified. A lot of the css stuff could also be removed if we bring in a ui library (surprised we don't have one already...) (As this seems to be a material look, we could probably add [MUI](https://mui.com/) as a UI lib)

Please also add a PBi for Tech Debt to improve the forms with some of the changes highlighted below .
(Side Note: I know its impossible sometimes but please keep PRs a bit smaller 😅)

Approved 👍

File: src/components/are-you-stuck/components/busi-booking/index.tsx (hunk 1 above)
Comment (ozair-ashfaque) [reply src/components/are-you-stuck/components/busi-booking/index.tsx:19]: Hi Jack, I will create a separate PBI for it, I am not sure why it has been showing in this branch. Thanks for pointing out

File: src/components/bookingFormFormik/formData/index.tsx (hunk 2 above)
Comment (ozair-ashfaque) [reply src/components/bookingFormFormik/formData/index.tsx:2]: I will do that.

File: src/components/bookingFormFormik/formData/index.tsx (hunk 3 above)
Comment (ozair-ashfaque) [reply src/components/bookingFormFormik/formData/index.tsx:40]: I will remove that one as well.

File: src/components/bookingForm/index.tsx (hunk 4 above)
Comment (ozair-ashfaque) [reply src/components/bookingForm/index.tsx:5]: Not really I will remove it.

File: src/components/bookingFormFormik/index.tsx (hunk 5 above)
Comment (ozair-ashfaque) [reply src/components/bookingFormFormik/index.tsx:57]: Haha, I will create a PBI for it as well

File: src/components/bookingFormFormik/index.tsx (hunk 8 above)
Comment (ozair-ashfaque) [reply src/components/bookingFormFormik/index.tsx:115]: I will add new PBI for refactoring it.

**Review by ozair-ashfaque — COMMENTED** [review]
I will create a PBI for techDebt

**Pull request merged**